readalongs/web_api.py

```diff
@@ -203,5 +204,174 @@ def create_grammar(xml):
     return dict_data, fsg_data, text_data
 
 
+class ConvertRequest(BaseModel):
+    """Convert Request contains the RAS-processed XML and SMIL alignments"""
+
+    audio_duration: float = Field(
+        example=2.01,
+        gt=0.0,
+        title="The duration of the audio used to create the alignment, in seconds.",
+    )
+
+    output_format: str = Field(
+        example="TextGrid",
+        regex="^(?i)(eaf|TextGrid|srt|vtt)$",
+        title="Format to convert to, one of TextGrid (Praat), eaf (ELAN), srt (SRT subtitles), or vtt (VTT subtitles).",
+    )
+
+    xml: str = Field(
+        title="The processed_xml returned by /assemble.",
+        example=dedent(
+            """\
+            <?xml version='1.0' encoding='utf-8'?>
+            <TEI>
+                <text xml:lang="dan" fallback-langs="und" id="t0">
+                    <body id="t0b0">
+                        <div type="page" id="t0b0d0">
+                            <p id="t0b0d0p0">
+                                <s id="t0b0d0p0s0"><w id="t0b0d0p0s0w0" ARPABET="HH EH Y">hej</w> <w id="t0b0d0p0s0w1" ARPABET="V Y D EH N">verden</w></s>
+                            </p>
+                        </div>
+                    </body>
+                </text>
+            </TEI>"""
+        ),
+    )
+
+    smil: str = Field(
+        title="The result of aligning xml to the audio with SoundSwallower(.js)",
+        example=dedent(
+            """\
+            <smil xmlns="http://www.w3.org/ns/SMIL" version="3.0">
+                <body>
+                    <par id="par-t0b0d0p0s0w0">
+                        <text src="hej-verden.xml#t0b0d0p0s0w0"/>
+                        <audio src="hej-verden.mp3" clipBegin="0.14" clipEnd="0.78"/>
+                    </par>
+                    <par id="par-t0b0d0p0s0w1">
+                        <text src="hej-verden.xml#t0b0d0p0s0w1"/>
+                        <audio src="hej-verden.mp3" clipBegin="0.78" clipEnd="1.89"/>
+                    </par>
+                </body>
+            </smil>"""
+        ),
+    )
+
+
+class ConvertResponse(BaseModel):
```

Comment:
Hm - I see why you did this - because some responses included two files, but I think we might actually want to design this a little differently. FastAPI has a built-in FileResponse object (https://fastapi.tiangolo.com/advanced/custom-response/?h=file#fileresponse) which takes care of sending appropriate headers along with the response (which we don't fully implement here). I think we should just parameterize the request for subtitles to return either a sentence or word response. More to come!

Reply:
Well, it turns out I cannot use a FileResponse, because that asynchronously reads the file to send it, but I wipe the file out as the endpoint end when I clean up the tempdir, as we promise in our data privacy policy, so I'll use a Response instead like this:
```
            return Response(
                slurp_file(prefix + ".TextGrid"),
                media_type="text/plain",
                headers={"Content-Disposition": "filename=aligned.TextGrid"},
            )
```

Reply:
I've also tested with `headers={"Content-Disposition": "attachment; filename=aligned.TextGrid"},` and it returns a download link with the contents, downloading to that filename, but I don't know if that's better for our use cases or not.

Reply:
Alternately if you don't want to read the whole file (but the files are small enough that I think this is a non-issue) you can do this:  https://fastapi.tiangolo.com/advanced/custom-response/#using-streamingresponse-with-file-like-objects

Reply:
Yeah, I was looking at StreamingResponse too, but I have to read it all into memory before I delete off the disk, so again there is not benefit in trying to do anything async optimized. The files are quite small, so I think whole-file fully sync is file.